src/main/java/ua/lviv/iot/novaPoshtaAPI/datastorage/dal/ParcelFileStore.java

```
@@ -0,0 +1,137 @@
+            } else {
+                scanner.nextLine();
+                isFirst = false;
+            }
+        }
+        return resultParcels;
+    }
+
+    private Parcel fillParcel(List<String> values) throws ParseException {
+        Parcel parcel = new Parcel();
+        int index = 0;
+        for (String value : values) {
+            switch (index) {
+                case 0 -> parcel.setParcelId(Long.parseLong(value));
+                case 1 -> parcel.setWeightInKgs(Float.parseFloat(value));
+                case 2 -> parcel.setHeightInCm(Float.parseFloat(value));
+                case 3 -> parcel.setWidthInCm(Float.parseFloat(value));
+                case 4 -> parcel.setLengthInCm(Float.parseFloat(value));
+                case 5 -> parcel.setOrigin(value);
+                case 6 -> parcel.setDestination(value);
+                case 7 -> parcel.setLocation(value);
+                case 8 -> parcel.setDateSent(new SimpleDateFormat("yyyy-MM-dd").parse(value));
+            }
+            index++;
+        }
+        return parcel;
+    }
+
+    public void saveParcels(final List<Parcel> parcels, boolean isTest) {
+        String date = Util.getDateNow();
+
+        File directory = new File("res");
+        if (!directory.exists()) {
+            directory.mkdir();
+        }
+
+        String testPath = "";
+        if (isTest) {
+            testPath = "test\\";
+        }
+        directory = new File("res\\test");
+        if (!directory.exists()) {
+            directory.mkdir();
+        }
+
+
+        File file = new File("res\\" + testPath + "parcel-" + date + ".csv");
+        try (Writer writer = new OutputStreamWriter(new FileOutputStream(file), StandardCharsets.UTF_8);) {
+            writer.write(parcels.get(0).getHeaders() + "\n");
+            for (Parcel parcel : parcels) {
+                writer.write(parcel.toCSV() + "\n");
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
```

do not suppress exceptions - your code will not work if it occurs